esp-hal-common/Cargo.toml

```diff
@@ -46,14 +47,10 @@ esp32s2 = { version = "0.3.0",  optional = true }
 esp32s3 = { version = "0.3.0",  optional = true }
 
 [features]
-esp32   = ["esp32/rt"  , "procmacros/xtensa", "multi_core" , "xtensa-lx-rt/esp32",   "xtensa-lx/esp32",   "critical-section/restore-state-u32"]
-esp32c3 = ["esp32c3/rt", "procmacros/riscv" , "single_core", "riscv", "riscv-atomic-emulation-trap",      "critical-section/restore-state-u8"]
-esp32s2 = ["esp32s2/rt", "procmacros/xtensa", "single_core", "xtensa-lx-rt/esp32s2", "xtensa-lx/esp32s2", "critical-section/restore-state-u32"]
-esp32s3 = ["esp32s3/rt", "procmacros/xtensa", "multi_core" , "xtensa-lx-rt/esp32s3", "xtensa-lx/esp32s3", "critical-section/restore-state-u32"]
-
-# Core Count (should not be enabled directly, but instead by a PAC's feature)
-single_core = []
-multi_core  = []
+esp32   = ["esp32/rt"  , "procmacros/xtensa", "xtensa-lx-rt/esp32",   "xtensa-lx/esp32",   "critical-section/restore-state-u32", "lock_api"]
```

Comment:
Can the `procmacros/xtensa` and `procmacros/riscv` features be eliminated too? I added these features because `target_arch` doesn't work with proc macros.

Reply:
I don't think this is possible unfortunately, but I could be wrong. Though a little outdated [this comment](https://github.com/rust-lang/cargo/issues/5499#issuecomment-533935297) indicates it is not possible.

Reply:
What about removing the features from procmacros and using the cfg(riscv) or cfg(xtensa) inside procmacros?